source/colors.js

```diff
@@ -0,0 +1,237 @@
+
```

Comment:
In Javascript, the order of object fields is not defined. I changed the structure so that the items we would like to be in a particular order (ie, "blue" first then "green" then "teal", etc, and "core" first then "10" then "20" etc within each palette) are represented by arrays, where the order of the elements is significant.

Also, I had thought to include the synonyms ("warm white", "grey", etc) into this structure, although they will not appear in the JSON format of the output -- I haven't done that though as part of this PR.

The template to transform into the JSON is very simple, and using a template rather than just stringify also allows us to set indentation and layout within the JSON file, which isn't significant for mechanical use of the file but can make it look more readable by the human eye.